subsys/bluetooth/host/iso.c

```diff
@@ -905,49 +873,27 @@ static struct net_buf *hci_le_set_cig_params(struct bt_iso_create_param *param)
 
 	memset(req, 0, sizeof(*req));
 
-	req->cig_id = param->conns[0]->iso.cig_id;
-	if (param->chans[0]->qos->tx) {
-		sys_put_le24(param->chans[0]->qos->tx->interval, req->m_interval);
-		req->m_latency = sys_cpu_to_le16(param->chans[0]->qos->tx->latency);
-	} else {
-		/* Use RX values if TX is disabled.
-		 * If TX is disabled then the values don't have any meaning, as
-		 * we will create a new CIG in case we want to change the
-		 * directions, but will need to be set to the fields to some
-		 * valid (nonzero) values for the controller to accept them.
-		 */
-		sys_put_le24(param->chans[0]->qos->rx->interval, req->m_interval);
-		req->m_latency = sys_cpu_to_le16(param->chans[0]->qos->rx->latency);
-	}
+	req->cig_id = cig->id;
+	req->m_latency = sys_cpu_to_le16(param->latency);
+	req->s_latency = sys_cpu_to_le16(param->latency);
+	sys_put_le24(param->interval, req->m_interval);
+	sys_put_le24(param->interval, req->s_interval);
 
-	if (param->chans[0]->qos->rx) {
-		sys_put_le24(param->chans[0]->qos->rx->interval, req->s_interval);
-		req->s_latency = sys_cpu_to_le16(param->chans[0]->qos->rx->latency);
-	} else {
-		/* Use TX values if RX is disabled.
-		 * If RX is disabled then the values don't have any meaning, as
-		 * we will create a new CIG in case we want to change the
-		 * directions, but will need to be set to the fields to some
-		 * valid (nonzero) values for the controller to accept them.
-		 */
-		sys_put_le24(param->chans[0]->qos->tx->interval, req->s_interval);
-		req->s_latency = sys_cpu_to_le16(param->chans[0]->qos->tx->latency);
-	}
-
-	req->sca = param->chans[0]->qos->sca;
-	req->packing = param->chans[0]->qos->packing;
-	req->framing = param->chans[0]->qos->framing;
-	req->num_cis = param->num_conns;
+	req->sca = param->sca;
+	req->packing = param->packing;
+	req->framing = param->framing;
+	req->num_cis = param->num_cis;
 
 	/* Program the cis parameters */
-	for (i = 0; i < param->num_conns; i++) {
-		struct bt_iso_chan_qos *qos = param->chans[i]->qos;
+	for (i = 0; i < param->num_cis; i++) {
+		struct bt_iso_chan *cis = param->cis_channels[i];
+		struct bt_iso_chan_qos *qos = cis->qos;
 
-		cis = net_buf_add(buf, sizeof(*cis));
+		cis_param = net_buf_add(buf, sizeof(*cis_param));
 
-		memset(cis, 0, sizeof(*cis));
+		memset(cis_param, 0, sizeof(*cis_param));
 
-		cis->cis_id = param->conns[i]->iso.cis_id;
+		cis_param->cis_id = cis->conn->iso.cis_id;
 
 		if (!qos->tx && !qos->rx) {
 			BT_ERR("Both TX and RX QoS are disabled");
```

Comment:
Should there be a return here, to avoid errors further down?
(Return NULL?)

Reply:
Not sure what you mean, it already returns NULL:

```
		if (!qos->tx && !qos->rx) {
			BT_ERR("Both TX and RX QoS are disabled");
			net_buf_unref(buf);
			return NULL;
		}
````

Reply:
I got fooled by lines not shown in the diff ...